### PR TITLE
nodejs-week3

### PR DESCRIPTION
**Pull request by oleksandra-lavryk**

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -1,12 +1,166 @@
...
+const { where } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    let mealArrayToChange = await knex
```
Comment (jslambda) [inline src/backend/api/meals.js]: Is there any particular reason that you didn't use knex chainable methods (e.g. knex.select and knex.where)?

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -1,12 +1,166 @@
...
+    }
+
+    if (request.query["createdAfter"]) {
+      const createdAfter = new Date(request.query["createdAfter"]);
+      if (createdAfter != "Invalid Date") {
+        mealArrayToChange = mealArrayToChange.filter((item) => {
+          const itemDate = new Date(item.createdAt);
+          return itemDate > createdAfter;
+        });
+      } else {
```
Comment (jslambda) [inline src/backend/api/meals.js:60]: Please consider using knex chainable methods for selection and filtering.

File: src/backend/api/meals.js [hunk 3]
```diff
@@ -1,12 +1,166 @@
...
+});
+
+router.get("/:id", async (request, response) => {
+  try {
+    const mealId = Number(request.params.id);
+    if (isNaN(mealId)) {
+      response.status(400).json({ error: "Invalid data" });
+      return;
+    }
+    const meal = await knex("meal")
```
Comment (jslambda) [inline src/backend/api/meals.js:88]: It makes the code more readable to rename meal above to meals as it is an array.

File: src/backend/api/meals.js (hunk 1 above)
Comment (oleksandra-lavryk) [reply src/backend/api/meals.js]: @jslambda yes, for parameter “availableReservation” I need to do complex query with JOIN SUM and HAVING and this “having” doesn’t work in chainable methods. I tried to google that problem but did not find solution.  So I used knex.raw.

File: src/backend/api/meals.js (hunk 3 above)
Comment (oleksandra-lavryk) [reply src/backend/api/meals.js:100]: @jslambda 
 This way to make query not working in some reason. "Having" part does not act as expected. 
      
      queryToMeal = queryToMeal
        .sum({ totalReserved: "reservation.number_of_guests" })
        .join("reservation", "reservation.meal_id", "=", "meal.id")
        .groupBy("meal.id")
        .having('totalReserved', '<','meal.max_reservations'); // 0 objects return here

File: src/backend/api/meals.js (hunk 2 above)
Comment (oleksandra-lavryk) [reply src/backend/api/meals.js:60]: @jslambda
This way to make query not working in some reason. "Having" part does not act as expected.

  queryToMeal = queryToMeal
    .sum({ totalReserved: "reservation.number_of_guests" })
    .join("reservation", "reservation.meal_id", "=", "meal.id")
    .groupBy("meal.id")
    .having('totalReserved', '<','meal.max_reservations'); // 0 objects return here

**Pull request merged**